### PR TITLE
Move EndpointType to `types` package.

### PR DESCRIPTION
**Pull request by masnax**
The hardcoded `1.0` value for resources supplied by the project using microcluster will be removed by #142, which means we need a way to supply the endpoint prefix to client functions.

This PR moves the type definition for `EndpointType` into the API types package, and moves all internal implementations into the internal API types package. It is also renamed to `EndpointPrefix` for clarity.

Then, when #142 is merged, this value will be specified in the set of `Resources` supplied to microcluster, and can be passed into the `Query` method.

### CODE REVIEW

**Comment by masnax** [comment]
@mas-who Once this PR is merged, it should solve the issue you were facing when removing the now-extraneous `EndpointType`.

File: client/client.go [hunk 1]
```diff
@@ -22,11 +23,11 @@ func IsNotification(r *http.Request) bool {
 
 // Query is a helper for initiating a request on the /1.0 endpoint. This function should be used for all client
```
Comment (mas-who) [inline client/client.go:24]: I think the comment here is now outdated

**Review by mas-who — COMMENTED** [review]
Thanks for the change! I think we should get this merged before #142

File: client/client.go (hunk 1 above)
Comment (masnax) [reply client/client.go]: Good catch, thanks :)

File: example/client/client.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func ExtendedPostCmd(ctx context.Context, c *client.Client, data *types.Extended
 	defer cancel()
 
 	var outStr string
-	err := c.Query(queryCtx, "POST", api.NewURL().Path("extended"), data, &outStr)
+	err := c.Query(queryCtx, "POST", "1.0", api.NewURL().Path("extended"), data, &outStr)
```
Comment (roosterfish) [inline example/client/client.go:23]: Can we use the `types.ExtendedEndpoint` const here?

**Review by roosterfish — CHANGES_REQUESTED** [review]
Overall this looks good, just one small comment.

File: example/client/client.go (hunk 2 above)
Comment (masnax) [reply example/client/client.go:23]: No, since it's from an internal package. This hardcoded value should get removed by #142.

Comment (roosterfish) [reply example/client/client.go:23]: Wasn't it moved away from the internal package as part of https://github.com/canonical/microcluster/pull/155/commits/e7c64e38c8b4de6b9fd7cb902827e4005dc7af04?

Comment (masnax) [reply example/client/client.go:23]: The type was, so external projects can set it in their `Server` structs after #142.

But all internally defined constants are in the [internal package](https://github.com/canonical/microcluster/blob/e7c64e38c8b4de6b9fd7cb902827e4005dc7af04/internal/rest/types/server.go)

Comment (roosterfish) [reply example/client/client.go:23]: Thanks, clear now.

**Review by roosterfish — APPROVED** [review]

**Pull request merged**